run.sh

```diff
@@ -1,15 +1,57 @@
 #!/bin/bash
 set -e
 set -x
-export DEVPI_SERVERDIR=/mnt
-export DEVPI_CLIENTDIR=/tmp/devpi-client
-[[ -f $DEVPI_SERVERDIR/.serverversion ]] || initialize=yes
-
-devpi-server --start --host 0.0.0.0 --port 3141
-if [[ $initialize = yes ]]; then
-  devpi use http://localhost:3141
-  devpi login root --password=''
-  devpi user -m root password="${DEVPI_PASSWORD}"
-  devpi index -y -c public pypi_whitelist='*'
-fi
-tail -f $DEVPI_SERVERDIR/.xproc/devpi-server/xprocess.log
+
+# Default environment if none is provided
+: ${DEVPI_SERVERIP:=127.0.0.1}
+: ${DEVPI_SERVERPORT:=3143}
+: ${DEVPI_SERVERDIR:=/mnt}
+: ${DEVPI_CLIENTDIR:=/tmp/devpi-client}
+: ${DEVPI_CLIENTPASSWORD:=""}
+: ${DEVPI_LDAPCONFILE:=""}
+
+: ${NGINX_IP:=0.0.0.0}
+: ${NGINX_PORT:=3141}
+: ${NGINX_ENABLETLS:=false}
+: ${NGINX_TLSPORT:=3142}
+: ${NGINX_TLSPATH:=/opt/nginx/certs/default}
+: ${NGINX_TLSCERT:=devpi.crt}
+: ${NGINX_KEY:=devpi.key}
+
+function start_devpi_server () {
+    # Start the server
+    devpi-server --start --host ${DEVPI_SERVERIP} --port ${DEVPI_SERVERPORT} &> stdout
+    
+    # Configure the default client if necessary
+    if [[ ! -f ${DEVPI_SERVERDIR}/.serverversion ]]; then
+      devpi use http://${DEVPI_SERVERIP}:${DEVPI_SERVERPORT}
+      devpi login root --password=""
+      devpi user -m root password="${DEVPI_CLIENTPASSWORD}"
+      devpi index -y -c public pypi_whitelist='*'
+    fi
+}
+
+function start_nginx () {
+    nginx_template_file=/opt/nginx/nginx-tpl.conf
+
+    # Uncomment lines that enable TLS in template file
+    if [ ${NGINX_ENABLETLS} = true ]; then
+        sed -i "s/#{uncomment_if_tls_enabled}\s*//g" ${nginx_template_file}
+    fi
+
+    # Escape all \ & and / for every variable of the current env, and replace all
+    # ${var} occurences in nginx conf template file with the corresponding variable
+    # value
+    env | sed 's/[\%]/\\&/g;s/\([^=]*\)=\(.*\)/s%${\1}%\2%/' | sed -f- ${nginx_template_file} | tee /tmp/nginx.conf    
```

Comment:
I thought I matters because substitutions were not happening in the nginx template. But this is actually because with the new syntax `: ${var:=default_val}`, I need to export the variables.